packages/callhierarchy/package.json

```diff
@@ -41,7 +43,11 @@
     "test": "theiaext test"
   },
   "devDependencies": {
-    "@theia/ext-scripts": "^0.9.0"
+    "@theia/core": "^0.9.0",
```

Comment:
Why do we need to have it `devDependencies` and in `peerDependencies`?

Reply:
A script could be made to ensure that whatever is defined as peer dependencies will be exactly the same in the dev dependencies. Building a Theia application would work the way it is, but when building each extension from the main repo, lerna gets lost and doesn't build in topological order. This is fixed by having everything as dev deps.

Reply:
Interesting thing to note as well: peer dependencies aren't automatically pulled. So when making a Theia application, you will have to declare each extension that will be used in your application.

But this is not a big issue, since omitting extensions currently does not work as well.

From https://www.theia-ide.org/doc/Composing_Applications.html, the following configuration will not allow you to build a correct Theia application (it will fail once loaded in the browser):

```json
{
  "private": true,
  "dependencies": {
    "typescript": "latest",
    "@theia/typescript": "next",
    "@theia/navigator": "next",
    "@theia/terminal": "next",
    "@theia/outline-view": "next",
    "@theia/preferences": "next",
    "@theia/messages": "next",
    "@theia/git": "next",
    "@theia/file-search": "next",
    "@theia/markers": "next",
    "@theia/preview": "next",
    "@theia/callhierarchy": "next",
    "@theia/merge-conflicts": "next",
    "@theia/search-in-workspace": "next",
    "@theia/json": "next",
    "@theia/textmate-grammars": "next",
    "@theia/mini-browser": "next"
  },
  "devDependencies": {
    "@theia/cli": "next"
  }
}
```

To fix this, just add `"@theia/core": "next"`. Despite all other extensions depending on `@theia/core`, somehow it will not be pulled correctly if not explicitly defined.

Reply:
I guess my question here is: Is it a big issue for clients to explicitly declare each extension that will be used in their application?

Reply:
@marechal-p , another question, does the end user know all dependencies for his application he needs. He will need the "@theia/core", but may be he also need "@theia/json" or any other dependencies to have his application running

Reply:
> I guess my question here is: Is it a big issue for clients to explicitly declare each Theia extension that will be used in their application? (even transitive ones)

It should not be necessary for sure.

Reply:
Implementation details would be prod deps, whereas extensions relying on one another (extensions defining contribution point + extensions providing contributions for it) would reference themselves using peer dependencies.